libs/java/auth_core/src/main/java/com/yahoo/athenz/auth/impl/CertificateAuthority.java

```diff
@@ -99,106 +96,22 @@ public Principal authenticate(X509Certificate[] certs, StringBuilder errMsg) {
-                return null;
-            }
-
-            // fist we need to keep the role name in our object
-            
-            roles = new ArrayList<>();
-            roles.add(principalName);
-            
-            // now extract the email field
-            
-            List<String> emails = Crypto.extractX509CertEmails(x509Cert);
-            if (emails.isEmpty()) {
-                reportError("CertificateAuthority: Invalid role cert, no email SAN entry", errMsg);
-                return null;
-            }
-            String email = emails.get(0);
-            idx = email.indexOf('@');
-            if (idx == -1) {
-                reportError("CertificateAuthority: Invalid role cert, invalid email SAN entry", errMsg);
-                return null;
-            }
-            principalName = email.substring(0, idx);
-        }
-
-        // check to see if we have a role certificate where roles
-        // are presented as URIs in the SAN
 
-        List<String> uris = Crypto.extractX509CertURIs(x509Cert);
-        for (String uri : uris) {
-            if (!uri.toLowerCase().startsWith(ZTS_CERT_ROLE_URI)) {
-                continue;
-            }
-            if (roles == null) {
-                roles = new ArrayList<>();
-            }
-            final String roleUri = uri.substring(ZTS_CERT_ROLE_URI.length());
-            idx = roleUri.indexOf('/');
-            if (idx == -1) {
-                reportError("CertificateAuthority: Invalid role cert, invalid uri SAN entry", errMsg);
-                return null;
-            }
-            roles.add(roleUri.substring(0, idx) + ":role." + roleUri.substring(idx + 1));
-        }
-
-        if (excludeRoleCertificates && roles != null) {
-            reportError("CertificateAuthority: Role Certificates not allowed", errMsg);
-            return null;
-        }
-
-        // extract domain and service names from the name. We must have
-        // a valid service identity in the form domain.service
-
-        idx = principalName.lastIndexOf('.');
-        if (idx == -1) {
-            reportError("CertificateAuthority: Principal is not a valid service identity", errMsg);
-            return null;
-        }
-
-        String domain = principalName.substring(0, idx);
-        String name = principalName.substring(idx + 1);
-        
-        // all the role members in Athenz are normalized to lower case so we need to make
-        // sure our principal's name and domain are created with lower case as well
-        
-        SimplePrincipal principal = (SimplePrincipal) SimplePrincipal.create(domain.toLowerCase(),
-                name.toLowerCase(), x509Cert.toString(), this);
+        // create principal
+        X509Certificate x509Cert = certId.getX509Certificate();
+        SimplePrincipal principal = (SimplePrincipal) SimplePrincipal.create(certId.getDomain(), certId.getService().toLowerCase(), x509Cert.toString(), this);
```

Comment:
seems inconsistent that we lowercase service but not domain?

Reply:
Thank you for pointing out. `CertificateIdentityParser` should handle the lowercase. `toLowerCase()` is not needed here.
Fixed in 28698703 and also removed other unnecessary `toLowerCase()`.